site/src/hooks/schedule.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+import { useAppDispatch } from '../store/hooks';
+import { setCurrentWeek, setCurrentQuarter } from '../store/slices/scheduleSlice';
+import trpc from '../trpc';
+
+export function useSetSchedule() {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    // set current week
+    trpc.schedule.currentWeek.query().then((res) => {
+      const week = res.display.split(' • ')[0];
+      dispatch(setCurrentWeek(week));
+    });
+
+    // set current quarter
+    trpc.schedule.currentQuarter.query().then((res) => {
+      dispatch(setCurrentQuarter(res));
+    });
+  }, [dispatch]);
```

Comment:
If I recall correctly, this will actually run twice due to React Strict mode. If you want to prevent this, see the approach I take in `site/src/component/CoursePreview/CoursePreview.tsx`.

---
I also had the initial concern that this doesn't have any sort of guard for if the original value exists, though that doesn't really matter since this is called once at the root level.

Reply:
`useEffect` is double-called on first render because of react strict mode. The suggestion I had would be to use the "[set]loadTrigger" so that the double render only applies to `setLoadTrigger(true)`, and the other hooks will only get past the `if (!loadTrigger) return false` once.

> I initially had that, removed it b/c of the reasoning u mentioned, but can easily add it back.

No need; was just a thought